kefir.js.flow

```diff
@@ -0,0 +1,158 @@
+  debounce(n: number, options?: {immediate?: boolean}): Observable<V,E>;
+  mapErrors<E2>(fn: (error: E) => E2): Observable<V,E2>;
+  filterErrors(fn: (error: E) => any): Observable<V,E>;
+  takeErrors(n: number): Observable<V,E>;
+  ignoreValues(): Observable<*,E>;
+  ignoreErrors(): Observable<V,*>;
+  ignoreEnd(): Observable<V,E>;
+  beforeEnd<V2>(fn: () => V2): Observable<V|V2,E>;
+  slidingWindow(max: number, min?: number): Observable<V[],E>;
+  bufferWhile(predicate?: (value: V) => boolean, options?: {flushOnEnd?: boolean}): Observable<V[],E>;
+  transduce(transducer: any): Observable<any,E>;
+  withHandler<V2,E2>(handler: (emitter: Emitter<V2,E2>, event: Event<V,E>) => void): Observable<V2,E2>;
+
+  combine<V2,E2>(otherObs: Observable<V2,E2>): Observable<[V,V2],E|E2>;
+  combine<V2,V3,E2>(otherObs: Observable<V2,E2>, combinator: (v: V, v2: V2) => V3): Observable<V3,E|E2>;
+  zip<V2,E2>(otherObs: Observable<V2,E2>): Observable<[V,V2],E|E2>;
+  zip<V2,V3,E2>(otherObs: Observable<V2,E2>, combinator: (v: V, v2: V2) => V3): Observable<V3,E|E2>;
+  merge<V2,E2>(otherObs: Observable<V2,E2>): Observable<V|V2,E|E2>;
+  concat<V2,E2>(otherObs: Observable<V2,E2>): Observable<V|V2,E|E2>;
+
+  flatMap<V2,E2>(transform: (value: V) => Observable<V2,E2>): Observable<V2,E|E2>;
+  flatMapLatest<V2,E2>(transform: (value: V) => Observable<V2,E2>): Observable<V2,E|E2>;
+  flatMapFirst<V2,E2>(transform: (value: V) => Observable<V2,E2>): Observable<V2,E|E2>;
+  flatMapConcat<V2,E2>(transform: (value: V) => Observable<V2,E2>): Observable<V2,E|E2>;
+  flatMapLimit<V2,E2>(transform: (value: V) => Observable<V2,E2>, limit: number): Observable<V2,E|E2>;
+  flatMapErrors<V2,E2>(transform: (error: E) => Observable<V2,E2>): Observable<V|V2,E2>;
+
+  filterBy(otherObs: Observable<any,any>): Observable<V,E>;
+  skipUntilBy(otherObs: Observable<any,any>): Observable<V,E>;
+  takeUntilBy(otherObs: Observable<any,any>): Observable<V,E>;
+  bufferBy(otherObs: Observable<any,any>, options?: {flushOnEnd?: boolean}): Observable<V[],E>;
+  bufferWhileBy(otherObs: Observable<any,any>, options?: {flushOnEnd?: boolean, flushOnChange?: boolean}): Observable<V[],E>;
+  sampledBy(otherObs: Observable<any,any>): Observable<V,E>;
+  sampledBy<V2,V3>(otherObs: Observable<V2,any>, combinator: (obsValue: V, otherObsValue: V2) => V3): Observable<V3,E>;
+  bufferWithTimeOrCount(time: number, count: number, options?: {flushOnEnd: boolean}): Observable<V,E>;
+}
+
+declare class Pool<V,E=*> extends Observable<V,E> {
+  plug(s: Observable<V,E>): () => void;
+}
+
+declare class Stream<V,E=*> extends Observable<V,E> {
+}
+
+declare class Property<V,E=*> extends Observable<V,E> {
+}
+
+declare var Kefir: {
+  Observable: typeof Observable;
+  Pool: typeof Pool;
+  Stream: typeof Stream;
+  Property: typeof Property;
+
+  staticLand: {
+    Observable: {
+      ap: Function; //TODO replace Function with more specific types
```

Comment:
Just in case you're struggling: `<A,B,E1,E2>(Observable<(x: A) => B, E1>, Observable<A, E2>): Observable<B, E1 | E2>`